### ISSUE TITLE
PrimaryKeyRelatedField(many=False) triggers exception

### ISSUE THREAD

**Issue by paxnovem** [tomchristie/django-rest-framework#3042]
On Django REST framework version 3.1.3, when I try to create a `PrimaryKeyRelatedField` with the argument `many=False` I get an exception about an unexpected keyword argument.

Here is the serializer that I am using

```
class BarSerializer(serializers.HyperlinkedModelSerializer):
    foo = serializers.PrimaryKeyRelatedField(
        many=False,
        read_only=True
    )
    
    class Meta:
        model = Bar
        fields = ('id', 'url', 'foo', )
```

Which triggers the following traceback when I try to start up the test server

```
Traceback (most recent call last):
  File "C:\code\funder_base\api\urls.py", line 4, in <module>
    from bar.views import BarViewSet
  File "C:\code\bar\views.py", line 4, in <module>
    from bar.serializers import BarSerializer
  File "C:\code\bar\serializers.py", line 16, in <module>
    class BarSerializer(serializers.HyperlinkedModelSerializer):
  File "C:\code\bar\serializers.py", line 18, in BarSerializer
    foo = serializers.PrimaryKeyRelatedField(many=False, read_only=True)
  File "C:\packages\rest_framework\relations.py", line 45, in __init__
    super(RelatedField, self).__init__(**kwargs)
TypeError: __init__() got an unexpected keyword argument 'many'
```
No error is given if the value of `many` is set to `True` or if `many` is left out entirely.

**Comment by xordoquy** [comment]
This is because `many=True` will trigger a substitution of the field with a list field.
I do think this is something we should fix.

**Comment by xordoquy** [comment]
Ok, so, more explanation about what's going on.

You won't have any issue with `many=True` although it isn't part of any of the `__init__` argument.
The `many` attribute is filtered out by [`RelatedField.__new__`](https://github.com/tomchristie/django-rest-framework/blob/a1e0bae9da7d69967b6bf2340aeb448ce154d2c7/rest_framework/relations.py#L47).
As you can see, `many` is pop out.
This works for `many=True` because:
1. we call `many_init` during the `__new__` which in turn creates a `ManyRelatedField`. Note that `kwargs` don't have `many` since it's been popped.
2. `__new__` returns an instance of `ManyRelatedField` which doesn't inherit from `RelatedField`. According to the `__new__` documentation the `__init__` won't be called by the Python interpreter.

This doesn't work when `many=False`:
1. Because `__new__` returns an instance of `RelatedField`
2. And the Python interpreter calls the `__init__` after `__new__` with the same arguments the `__new__` had, meaning that `many=False` is still in the kwargs.

**Comment by tomchristie** [comment]
Closed by #3045.
